File: resource/cpp/com/template/Boat.cpp
#include "Boat.h"
void Boat::read(ByteBuffer* buffer)
{
		this->boat_id=buffer->readUTF();
		this->name=buffer->readUTF();
		this->des=buffer->readUTF();
		this->color=buffer->readShort();
		this->build_id=buffer->readUTF();
		short consume_goods_count=buffer->readShort();
		if(consume_goods_count>0)
		{
			for(int i=0;i<consume_goods_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->consume_goods.push_back(tmp);
			}
		}

		this->build_time=buffer->readLong();
		this->pos=buffer->readShort();
		this->weight=buffer->readLong();
		this->attack_type=buffer->readShort();
		this->attack_init=buffer->readFloat();
		this->life_init=buffer->readFloat();
		this->hit_init=buffer->readFloat();
		this->duck_init=buffer->readFloat();
		this->boom_init=buffer->readFloat();
		this->toughness_init=buffer->readFloat();
		this->qte_init=buffer->readFloat();
		short skill_id_count=buffer->readShort();
		if(skill_id_count>0)
		{
			for(int i=0;i<skill_id_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->skill_id.push_back(tmp);
			}
		}

		short passive_skill_id_count=buffer->readShort();
		if(passive_skill_id_count>0)
		{
			for(int i=0;i<passive_skill_id_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->passive_skill_id.push_back(tmp);
			}
		}

		this->attack_effect=buffer->readUTF();
		this->hit_attack=buffer->readUTF();
		this->attack_time=buffer->readInt();
		this->hit_time=buffer->readInt();
		this->shap_id=buffer->readUTF();
		this->icon_id=buffer->readUTF();
		this->attack_id=buffer->readUTF();
		this->scale=buffer->readFloat();
		this->effect=buffer->readUTF();
}

File: resource/java/com/template/Boat.java
package com.template;
import java.io.DataInputStream;
public class Boat implements TemplateInterface
{
	public String boat_id;
	public String name;
	public String des;
	public short color;
	public String build_id;
	public String[] consume_goods;
	public long build_time;
	public short pos;
	public long weight;
	public short attack_type;
	public float attack_init;
	public float life_init;
	public float hit_init;
	public float duck_init;
	public float boom_init;
	public float toughness_init;
	public float qte_init;
	public String[] skill_id;
	public String[] passive_skill_id;
	public String attack_effect;
	public String hit_attack;
	public int attack_time;
	public int hit_time;
	public String shap_id;
	public String icon_id;
	public String attack_id;
	public float scale;
	public String effect;
	public void read(DataInputStream in)throws Exception
	{
		boat_id=in.readUTF();
		name=in.readUTF();
		des=in.readUTF();
		color=in.readShort();
		build_id=in.readUTF();
		short consume_goods_count=in.readShort();
		if(consume_goods_count>0)
		{
			consume_goods=new String[consume_goods_count];
			for(int i=0;i<consume_goods_count;i++)
			{
				consume_goods[i]=in.readUTF();
			}
		}

		build_time=in.readLong();
		pos=in.readShort();
		weight=in.readLong();
		attack_type=in.readShort();
		attack_init=in.readFloat();
		life_init=in.readFloat();
		hit_init=in.readFloat();
		duck_init=in.readFloat();
		boom_init=in.readFloat();
		toughness_init=in.readFloat();
		qte_init=in.readFloat();
		short skill_id_count=in.readShort();
		if(skill_id_count>0)
		{
			skill_id=new String[skill_id_count];
			for(int i=0;i<skill_id_count;i++)
			{
				skill_id[i]=in.readUTF();
			}
		}

		short passive_skill_id_count=in.readShort();
		if(passive_skill_id_count>0)
		{
			passive_skill_id=new String[passive_skill_id_count];
			for(int i=0;i<passive_skill_id_count;i++)
			{
				passive_skill_id[i]=in.readUTF();
			}
		}

		attack_effect=in.readUTF();
		hit_attack=in.readUTF();
		attack_time=in.readInt();
		hit_time=in.readInt();
		shap_id=in.readUTF();
		icon_id=in.readUTF();
		attack_id=in.readUTF();
		scale=in.readFloat();
		effect=in.readUTF();
	}
}
